data/World/Water Temple.json

```diff
@@ -12,7 +12,13 @@
                 # Ensure that the water level can be raised if it were to be lowered.
         },
         "locations": {
-            "Fairy Pot": "Raise_Water_Level and has_bottle and can_use(Longshot)"
+            "Fairy Pot": "Raise_Water_Level and has_bottle and can_use(Longshot)",
+            "Water Temple Main Room L2 Pot 1" : "at('Water Temple Middle Water Level', 'True') or
+                ((can_use(Iron_Boots) or (Progressive_Scale, 2)) and (can_use(Bow) or can_use(Hookshot)))", 
+            "Water Temple Main Room L2 Pot 2" : "at('Water Temple Middle Water Level', 'True') or
+                ((can_use(Iron_Boots) or (Progressive_Scale, 2)) and (can_use(Bow) or can_use(Hookshot)))",
+            "Water Temple L1 Torch Pot 1" : "at('Water Temple Dive', 'True') and (can_use(Iron_Boots) and (can_use(Hookshot) or can_play(Zeldas_Lullaby)))",
+            "Water Temple L1 Torch Pot 2" : "at('Water Temple Dive', 'True') and (can_use(Iron_Boots) and (can_use(Hookshot) or can_play(Zeldas_Lullaby)))"
```

Comment:
I'm not sure why these pots aren't just in the water temple dive region, rather than at(true).

The way this is written always requires iron boots, but the trick to longshot the torches at the bottom avoids needing them. It should be "(irons and hook) or zl" not "irons (and hook or zl)".

(I didn't try very hard but I'm convinced a bombchu method is possible lol.)